Added lesson08 and lesson09 exercises

RimleeTalukdar44 opened the pull request:
Added the below-
1. circle.py
2. test_circle.py
3. mailroom_oop.py

RimleeTalukdar44 committed:
Added circle.py & test_circle.py

RimleeTalukdar44 committed:
Added mailroom_oop.py

PythonCHB commented on students/RimleeT/lesson08/circle.py:
```diff
@@ -0,0 +1,61 @@
+            return Circle(self.radius * other)
+
+    __rmul__ = __mul__
+    __iadd__ = __add__
```
careful here -- "iadd" means "in-place add" -- when defined for mutable objects, such as Circle, it should alter the existing instance (self), rather than make a new one, so it needs to be defined differently than __add__. A test for this would be:

```
c = Circle(5)
c2 = c  # a new name for the same circle
c + 4
assert c2.radius == 9  # the original Circle should have been changed 
```

PythonCHB commented on students/RimleeT/lesson08/test_circle.py:
```diff
@@ -0,0 +1,127 @@
+
+def test_sphere_volume():
+    a = Sphere(4)
+    assert math.isclose(268.08, a.volume, rel_tol=1e-4)
```
you need some more Sphere tests -maybe not everything, but does the from_diameter constructor work? Does addition and all that work?

PythonCHB commented on students/RimleeT/lesson08/circle.py:
```diff
@@ -0,0 +1,61 @@
+        return self.radius == other.radius
+
+    def __mul__(self, other):
+        if isinstance(other, int):
```
careful here: this will only work for integers, not floating point numbers -- not as useful.

PythonCHB reviewed (changes requested):
Circle is mostly done -- we'll give you credit -- but do look more at it if you can.

OO mailroom doesn't appear to have anything OO in it -- did you foget to commit those changes?

RimleeTalukdar44 committed:
cli_main.py

RimleeTalukdar44 committed:
Added donor_models.py

RimleeTalukdar44 commented:
Hi Chris,

Apologies ! I made a blunder added the old file by mistake. I have
committed the below files-
1. cli_main.py
2. donor_models.py

Kindly review and grade these files.I could not make a pull request , it
just shows 'View pull request' when I try to do one. Can you discard the
previous pull request so that I can make a new one ?
I am working on the circle.py. Will commit it soon.

Thanks you very much.


Regards,
Rimlee


On Sun, Jun 16, 2019 at 8:54 AM Christopher H.Barker, PhD <
notifications@github.com> wrote:

> *@PythonCHB* requested changes on this pull request.
>
> Circle is mostly done -- we'll give you credit -- but do look more at it
> if you can.
>
> OO mailroom doesn't appear to have anything OO in it -- did you foget to
> commit those changes?
> ------------------------------
>
> In students/RimleeT/lesson08/test_circle.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/SP_2019_210A_classroom/pull/192#discussion_r294090286>
> :
>
> > +def test_operation_equal():
> +    a = Circle(3)
> +    a += Circle(1)
> +    assert 4 == a.radius
> +
> +    a *= 2
> +    assert 8 == a.radius
> +
> +def test_sphere_str():
> +    a = Sphere(2)
> +    assert "Sphere with radius: 2.000000" == str(a)
> +    assert "Sphere(2)" == repr(a)
> +
> +def test_sphere_volume():
> +    a = Sphere(4)
> +    assert math.isclose(268.08, a.volume, rel_tol=1e-4)
>
> you need some more Sphere tests -maybe not everything, but does the
> from_diameter constructor work? Does addition and all that work?
> ------------------------------
>
> In students/RimleeT/lesson08/circle.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/SP_2019_210A_classroom/pull/192#discussion_r294090298>
> :
>
> > +        return "Circle({})".format(round(self.radius))
> +
> +    def __add__(self, other):
> +        return Circle(self.radius + other.radius)
> +
> +    def __gt__(self, other):
> +        return self.radius > other.radius
> +
> +    def __lt__(self, other):
> +        return self.radius < other.radius
> +
> +    def __eq__(self, other):
> +        return self.radius == other.radius
> +
> +    def __mul__(self, other):
> +        if isinstance(other, int):
>
> careful here: this will only work for integers, not floating point numbers
> -- not as useful.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/UWPCE-PythonCert-ClassRepos/SP_2019_210A_classroom/pull/192?email_source=notifications&email_token=AG6WYX3UIH66F6QUR2IGWV3P2ZOZ7A5CNFSM4HYQB3Z2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB3VFMQI#pullrequestreview-250238529>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AG6WYX2ANG3IHGQM3QA6V7DP2ZOZ7ANCNFSM4HYQB3ZQ>
> .
>

RimleeTalukdar44 commented:
Hey Rimlee,

When you continue to commit/push files, it will add it to your existing PR automatically. If you want to remove any files, you can do “git rm filename” and then commit/push changes to remove it from the PR.

Let me know if you have any troubles with it!

Thanks,
Ron

Sent from my iPhone

On Jun 16, 2019, at 13:02, rimlee talukdar <rimlee.talukdar@gmail.com<mailto:rimlee.talukdar@gmail.com>> wrote:

Hi Chris,

Apologies ! I made a blunder added the old file by mistake. I have committed the below files-
1. cli_main.py
2. donor_models.py

Kindly review and grade these files.I could not make a pull request , it just shows 'View pull request' when I try to do one. Can you discard the previous pull request so that I can make a new one ?
I am working on the circle.py. Will commit it soon.

Thanks you very much.


Regards,
Rimlee


On Sun, Jun 16, 2019 at 8:54 AM Christopher H.Barker, PhD <notifications@github.com<mailto:notifications@github.com>> wrote:

@PythonCHB requested changes on this pull request.

Circle is mostly done -- we'll give you credit -- but do look more at it if you can.

OO mailroom doesn't appear to have anything OO in it -- did you foget to commit those changes?

________________________________

In students/RimleeT/lesson08/test_circle.py<https://github.com/UWPCE-PythonCert-ClassRepos/SP_2019_210A_classroom/pull/192#discussion_r294090286>:

> +def test_operation_equal():
+    a = Circle(3)
+    a += Circle(1)
+    assert 4 == a.radius
+
+    a *= 2
+    assert 8 == a.radius
+
+def test_sphere_str():
+    a = Sphere(2)
+    assert "Sphere with radius: 2.000000" == str(a)
+    assert "Sphere(2)" == repr(a)
+
+def test_sphere_volume():
+    a = Sphere(4)
+    assert math.isclose(268.08, a.volume, rel_tol=1e-4)


you need some more Sphere tests -maybe not everything, but does the from_diameter constructor work? Does addition and all that work?

________________________________

In students/RimleeT/lesson08/circle.py<https://github.com/UWPCE-PythonCert-ClassRepos/SP_2019_210A_classroom/pull/192#discussion_r294090298>:

> +        return "Circle({})".format(round(self.radius))
+
+    def __add__(self, other):
+        return Circle(self.radius + other.radius)
+
+    def __gt__(self, other):
+        return self.radius > other.radius
+
+    def __lt__(self, other):
+        return self.radius < other.radius
+
+    def __eq__(self, other):
+        return self.radius == other.radius
+
+    def __mul__(self, other):
+        if isinstance(other, int):


careful here: this will only work for integers, not floating point numbers -- not as useful.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/UWPCE-PythonCert-ClassRepos/SP_2019_210A_classroom/pull/192?email_source=notifications&email_token=AG6WYX3UIH66F6QUR2IGWV3P2ZOZ7A5CNFSM4HYQB3Z2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB3VFMQI#pullrequestreview-250238529>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AG6WYX2ANG3IHGQM3QA6V7DP2ZOZ7ANCNFSM4HYQB3ZQ>.

RimleeTalukdar44 committed:
deleted old mailroom file

RimleeTalukdar44 commented:
> Hi Chris,
> 
> Apologies ! I made a blunder added the old file by mistake. I have
> committed the below files-
> 1. cli_main.py
> 2. donor_models.py
> 
> Kindly review and grade these files.I could not make a pull request , it
> just shows 'View pull request' when I try to do one. Can you discard the
> previous pull request so that I can make a new one ?
> I am working on the circle.py. Will commit it soon.
> 
> Thanks you very much.
> 
> 
> Regards,
> Rimlee
> …

RimleeTalukdar44 commented:
Hi Ron,

I removed the old mailroom file which I added by mistake. Added the below
files and made a pull request again (#193)
1. cli_main.py
2. donor_models.py

Kindly review the above.I could not get time to write test cases for
mailroom oop.
Thanks a lot.

Regards,
Rimlee


On Sun, Jun 16, 2019 at 11:06 AM Raghav Nair <nairraghav@hotmail.com> wrote:

> Hey Rimlee,
>
> When you continue to commit/push files, it will add it to your existing PR
> automatically. If you want to remove any files, you can do “git rm
> filename” and then commit/push changes to remove it from the PR.
>
> Let me know if you have any troubles with it!
>
> Thanks,
> Ron
>
> Sent from my iPhone
>
> On Jun 16, 2019, at 13:02, rimlee talukdar <rimlee.talukdar@gmail.com>
> wrote:
>
> Hi Chris,
>
> Apologies ! I made a blunder added the old file by mistake. I have
> committed the below files-
> 1. cli_main.py
> 2. donor_models.py
>
> Kindly review and grade these files.I could not make a pull request , it
> just shows 'View pull request' when I try to do one. Can you discard the
> previous pull request so that I can make a new one ?
> I am working on the circle.py. Will commit it soon.
>
> Thanks you very much.
>
>
> Regards,
> Rimlee
>
>
> On Sun, Jun 16, 2019 at 8:54 AM Christopher H.Barker, PhD <
> notifications@github.com> wrote:
>
>> *@PythonCHB* requested changes on this pull request.
>>
>> Circle is mostly done -- we'll give you credit -- but do look more at it
>> if you can.
>>
>> OO mailroom doesn't appear to have anything OO in it -- did you foget to
>> commit those changes?
>> ------------------------------
>>
>> In students/RimleeT/lesson08/test_circle.py
>> <https://github.com/UWPCE-PythonCert-ClassRepos/SP_2019_210A_classroom/pull/192#discussion_r294090286>
>> :
>>
>> > +def test_operation_equal():
>> +    a = Circle(3)
>> +    a += Circle(1)
>> +    assert 4 == a.radius
>> +
>> +    a *= 2
>> +    assert 8 == a.radius
>> +
>> +def test_sphere_str():
>> +    a = Sphere(2)
>> +    assert "Sphere with radius: 2.000000" == str(a)
>> +    assert "Sphere(2)" == repr(a)
>> +
>> +def test_sphere_volume():
>> +    a = Sphere(4)
>> +    assert math.isclose(268.08, a.volume, rel_tol=1e-4)
>>
>> you need some more Sphere tests -maybe not everything, but does the
>> from_diameter constructor work? Does addition and all that work?
>> ------------------------------
>>
>> In students/RimleeT/lesson08/circle.py
>> <https://github.com/UWPCE-PythonCert-ClassRepos/SP_2019_210A_classroom/pull/192#discussion_r294090298>
>> :
>>
>> > +        return "Circle({})".format(round(self.radius))
>> +
>> +    def __add__(self, other):
>> +        return Circle(self.radius + other.radius)
>> +
>> +    def __gt__(self, other):
>> +        return self.radius > other.radius
>> +
>> +    def __lt__(self, other):
>> +        return self.radius < other.radius
>> +
>> +    def __eq__(self, other):
>> +        return self.radius == other.radius
>> +
>> +    def __mul__(self, other):
>> +        if isinstance(other, int):
>>
>> careful here: this will only work for integers, not floating point
>> numbers -- not as useful.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/UWPCE-PythonCert-ClassRepos/SP_2019_210A_classroom/pull/192?email_source=notifications&email_token=AG6WYX3UIH66F6QUR2IGWV3P2ZOZ7A5CNFSM4HYQB3Z2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB3VFMQI#pullrequestreview-250238529>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AG6WYX2ANG3IHGQM3QA6V7DP2ZOZ7ANCNFSM4HYQB3ZQ>
>> .
>>
>

RimleeTalukdar44 commented:
Hi Ron,

Gentle reminder!
Kindly review the pull request #193

Thanks and regards,
Rimlee

> On Jun 16, 2019, at 11:38 AM, rimlee talukdar <rimlee.talukdar@gmail.com> wrote:
> 
> Hi Ron,
> 
> I removed the old mailroom file which I added by mistake. Added the below files and made a pull request again (#193)
> 1. cli_main.py
> 2. donor_models.py
> 
> Kindly review the above.I could not get time to write test cases for mailroom oop. 
> Thanks a lot.
> 
> Regards,
> Rimlee
> 
> 
>> On Sun, Jun 16, 2019 at 11:06 AM Raghav Nair <nairraghav@hotmail.com> wrote:
>> Hey Rimlee, 
>> 
>> When you continue to commit/push files, it will add it to your existing PR automatically. If you want to remove any files, you can do “git rm filename” and then commit/push changes to remove it from the PR.
>> 
>> Let me know if you have any troubles with it! 
>> 
>> Thanks,
>> Ron
>> 
>> Sent from my iPhone
>> 
>> On Jun 16, 2019, at 13:02, rimlee talukdar <rimlee.talukdar@gmail.com> wrote:
>> 
>>> Hi Chris,
>>> 
>>> Apologies ! I made a blunder added the old file by mistake. I have committed the below files-
>>> 1. cli_main.py
>>> 2. donor_models.py
>>> 
>>> Kindly review and grade these files.I could not make a pull request , it just shows 'View pull request' when I try to do one. Can you discard the previous pull request so that I can make a new one ?
>>> I am working on the circle.py. Will commit it soon.
>>> 
>>> Thanks you very much.
>>> 
>>> 
>>> Regards,
>>> Rimlee
>>> 
>>> 
>>>> On Sun, Jun 16, 2019 at 8:54 AM Christopher H.Barker, PhD <notifications@github.com> wrote:
>>>> @PythonCHB requested changes on this pull request.
>>>> 
>>>> Circle is mostly done -- we'll give you credit -- but do look more at it if you can.
>>>> 
>>>> OO mailroom doesn't appear to have anything OO in it -- did you foget to commit those changes?
>>>> 
>>>> In students/RimleeT/lesson08/test_circle.py:
>>>> 
>>>> > +def test_operation_equal():
>>>> +    a = Circle(3)
>>>> +    a += Circle(1)
>>>> +    assert 4 == a.radius
>>>> +
>>>> +    a *= 2
>>>> +    assert 8 == a.radius
>>>> +
>>>> +def test_sphere_str():
>>>> +    a = Sphere(2)
>>>> +    assert "Sphere with radius: 2.000000" == str(a)
>>>> +    assert "Sphere(2)" == repr(a)
>>>> +
>>>> +def test_sphere_volume():
>>>> +    a = Sphere(4)
>>>> +    assert math.isclose(268.08, a.volume, rel_tol=1e-4)
>>>> you need some more Sphere tests -maybe not everything, but does the from_diameter constructor work? Does addition and all that work?
>>>> 
>>>> In students/RimleeT/lesson08/circle.py:
>>>> 
>>>> > +        return "Circle({})".format(round(self.radius))
>>>> +
>>>> +    def __add__(self, other):
>>>> +        return Circle(self.radius + other.radius)
>>>> +
>>>> +    def __gt__(self, other):
>>>> +        return self.radius > other.radius
>>>> +
>>>> +    def __lt__(self, other):
>>>> +        return self.radius < other.radius
>>>> +
>>>> +    def __eq__(self, other):
>>>> +        return self.radius == other.radius
>>>> +
>>>> +    def __mul__(self, other):
>>>> +        if isinstance(other, int):
>>>> careful here: this will only work for integers, not floating point numbers -- not as useful.
>>>> 
>>>> —
>>>> You are receiving this because you authored the thread.
>>>> Reply to this email directly, view it on GitHub, or mute the thread.
